components/merkledb/src/proof_map_index/mod.rs

```diff
@@ -201,7 +201,7 @@ pub type RawProofMapIndex<T, K, V> = ProofMapIndex<T, K, V, Raw>;
 impl<T, K, V, KeyMode> ProofMapIndex<T, K, V, KeyMode>
 where
     T: RawAccess,
-    K: BinaryKey,
+    K: BinaryKey + ToOwned + ?Sized,
```

Comment:
Nit: `BinaryKey + ToOwned` looks unnecessary since `BinaryKey` is a subtrait of `ToOwned`.